### ISSUE TITLE
Use in-memory DB to potentially speed up the tests

### ISSUE THREAD

**Issue by clemens-msupply** [openmsupply/remote-server#979]
Try to speedup tests by running sqlite in-memory.

To make this work `test_db::setup` could return a ready to use ConnectionManager. This is needed to not drop the just initialized in-memory DB.

In memory dbs can also be shared (if needed):
https://sqlite.org/forum/info/cf7424bc7b4f37e1

**Comment by andreievg** [comment]
> In memory dbs can also be shared (if needed):

How would you share something between tests, unless you have one tests that calls other test (it would be slower though since tests wont' run in parallel)

I think there are these use cases for testing while deving:

- 1 Creating a new test or modifying existing test (use ide to run just that test)
- 2 Making sure things are not broken when changing multiple files (this is the one that needs speed up, can just run with sqlite)
- 2 Before PR is submitted (need to do a full test for both features), would propose having two copies of the repo, one for pre PR tests (so that can work on another PR/branch while test is running)

I've just did a test, after compilation:
Sqlite test: 45 seconds
Postgres test: 60 seconds

I think we can speed up test by sorting out our mock test data (just use one 'base' central common data set, and all other test data is initiated with the new way .join on mock data in combination with inline init/edit).

I tried to do a quick test, but it's not straight forward
